### PR TITLE
Vitus/test changes

### PR DESCRIPTION
**Pull request by VitusAcabado**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- docs update for testing against marqo cloud server
- added logic for tests based on if multi-instance

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
image

### CODE REVIEW

File: tests/test_buildspec.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: 0.2
```
Comment (pandu-k) [inline tests/test_buildspec.yaml]: what is this file for?

File: tests/marqo_test.py [hunk 2]
```diff
@@ -5,14 +5,37 @@
...
+
+        marqo_server_version = Client(**cls.client_settings).get_marqo()["version"]
+        if marqo_server_version != py_marqo_support_version:
+            print(f"WARNING: supported py Marqo version and Marqo versions aren't the same!\n {marqo_server_version} != {py_marqo_support_version}")
+            print(f"MARQO SERVER VERSION -> {marqo_server_version}")
+            print(f"PY-MARQO SUPPORTED VERSION -> {py_marqo_support_version}")
+
+
+
+    def warm_request(f) -> None:
```
Comment (pandu-k) [inline tests/marqo_test.py]: add  docstring about what this does and when it should be used

Comment (pandu-k) [reply tests/marqo_test.py]: are you sure this runs the command 5 times? or does it execute once and do some no-op with the result?

Comment (pandu-k) [reply tests/marqo_test.py]: what might work is something like this:
```
def warm_request(func, *args, **kwargs):
    for i in range(5):
        func(*args, **kwargs) 
```

File: tests/test_buildspec.yaml (hunk 1 above)
Comment (VitusAcabado) [reply tests/test_buildspec.yaml]: file used to run the tests in cloud pipeline.
I can remove it if you want.

File: tests/marqo_test.py (hunk 2 above)
Comment (VitusAcabado) [reply tests/marqo_test.py]: yep the command runs it 5 times based on the logs, but I can modify the code as per your suggestion

File: tests/test_buildspec.yaml (hunk 1 above)
Comment (pandu-k) [reply tests/test_buildspec.yaml]: nah no stress. But a comment at the top to help understand what might be useful

File: tests/v0_tests/test_add_documents.py [hunk 3]
```diff
@@ -316,14 +324,26 @@ def test_update_documents(self):
...
         self.client.index(self.index_name_1).add_documents(documents=docs)
+
+        if self.IS_MULTI_INSTANCE:
+            time.sleep(1)
+
         assert 2 == self.client.index(self.index_name_1).get_stats()['numberOfDocuments']
         assert d1 == self.client.index(self.index_name_1).get_document(document_id='1234')
         assert d2 == self.client.index(self.index_name_1).get_document(document_id='5678')
+
+
```
Comment (pandu-k) [inline tests/v0_tests/test_add_documents.py:344]: Nit: 1 line between class methods

File: tests/v0_tests/test_add_documents.py [hunk 4]
```diff
@@ -447,10 +467,22 @@ def run():
     def test_add_lists_non_tensor(self):
         original_doc = {"d1": "blah", "_id": "1234", 'my list': ['tag-1', 'tag-2']}
         self.client.index(self.index_name_1).add_documents(documents=[original_doc], non_tensor_fields=['my list'])
+
+        if self.IS_MULTI_INSTANCE:
+            self.warm_request(self.client.index(self.index_name_1).search,
+                q='something', filter_string='my\ list:tag-1'
+            )
```
Comment (pandu-k) [inline tests/v0_tests/test_add_documents.py:474]: Kwargs need to be a dict:

{"k": "v" ...}

**Review by pandu-k — APPROVED** [review]

**Pull request merged**